golem-worker-service-base/src/service/worker/default.rs

```diff
@@ -85,24 +82,13 @@ pub trait WorkerService<AuthCtx> {
         auth_ctx: &AuthCtx,
     ) -> WorkerResult<()>;
 
-    // Accepts Json Params and Returns Json (with no type information)
+    // Accepts Vec<TypeAnnotatedValue> and returns TypeAnnotatedValue
     async fn invoke_and_await_function_json(
```

Comment:
should not it be invoke_and_await_function_json -> invoke_and_await_function_typed ?

Reply:
I don't like the `_typed` postfix because it's always typed so chose the `_json` one (previously we had both). This function is called when the REST API is called with JSON parameters, which are `TypeAnnotatedValue`'s.